lib/index.js

```diff
@@ -241,12 +245,60 @@ Socket.prototype._read = function (n) {}
  * Socket API
  */
 
-Socket.prototype.bind = function (addr) {
-  return this.bound[addr] = this._protect(nn.Bind( this.binding, addr ));
+Socket.prototype.bind = function (addr, fn) {
+  if (fn) return asyncBindWrapper.call(this, nn.asyncBind);
+  var ret;
+  this.sync('Bind', this, addr, ret);
```

Comment:
yea that's true `ret` must be undefined. So we could withhold the arg from being passed to `this.sync` and that should make it *just* `undefined` (at least declared) inside `this.sync`. 

it would work but look again at deasync helpers. `this.sync` is a transformed `function block()`, after getting passed to `deasyc(block)`.
```js 
/*  deasync helpers */
Socket.prototype.sync = deasync(block);
function block (nnMethod, ctx, opt, ret, done) {
  var throttle = setInterval(function(){
    if (ret !== undefined) {
      clearInterval(throttle);
      done();
    }
  }, 10);
  ret = ctx._protect(nn[nnMethod]( ctx.binding, opt ));
  switch (nnMethod) {
    case 'Bind'     : return ctx.bound[opt]       = ret;
    case 'Connect'  : return ctx.connected[opt]   = ret;
  }
}
function deasync (fn) {
  return function () {
    var done=false, err, res;
    fn.apply(this, Array.prototype.slice.apply(arguments).concat(cb));
    while (!done) nn.block();
    if (err) throw err;
    return res;
    function cb (e, r) { err=e; res=r; done=true; };
  }
}
```